new mechanics and some features

AAA1459 opened the pull request.

AAA1459 committed:
New mechanics and some features

AAA1459 committed:
some textures for testing #1

AAA1459 committed:
Delete Graphics directory

AAA1459 committed:
some textures for testing --A--

AAA1459 committed:
some textures for testing --B--

AAA1459 committed:
reset the version number to 1.0.0

Set the version number to 1.0.0 to prevent other mods get old values

AAA1459 committed:
tweaked the code about custom colorGrade

Fixed a bug, then got a new bug

AAA1459 committed:
Added "if Player's MaxDashes is 0" case

be like vanilla: if MaxDashes is 0, so the Madeline's Hair is red, not blue

AAA1459 committed:
Added "if Player's MaxDashes is 0" case

be like vanilla: if MaxDashes is 0, so the Madeline's Hair is red, not blue

AAA1459 committed:
Optimize some the codes about hair

and tweaked a few things, and Can get silhouette mode (can to check mini_guide.txt)

AAA1459 committed:
Simplify some codes

AAA1459 committed:
remove ModeValue of ConfigFlie, Add auto-hashValue

Use auto-hashValue to Compatible with CelesteNet, instead of ModeValue of ConfigFlie

Record the player's DashCount into hashValue, Made ColorGrade truly Compatible with CelesteNet

For alternatives to the removed ModeValue etc, see mini_guide.txt

AAA1459 committed:
Fixed some Error-Jump about ColorGrade

AAA1459 committed:
guide change test

AAA1459 committed:
guide change test 2

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Add files via upload

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

rhelmot commented on Code/SkinModHelperConfig.cs:
```diff
@@ -1,21 +1,48 @@
-        public string GetUniquePath() {
-            return SkinId.Replace('_', '/') + '/';
-        }
+    public class SkinModHelperConfig {
```
This is probably the most problematic change in this entire pull request - you cannot break the existing mods that are using SkinModHelper. This changes the schema in a backwards-incompatible way. If you want to make changes like this, you need to either make them backwards compatible, make a new mod, or convince everyone to change every single skin mod.

rhelmot commented on everest.yaml:
```diff
@@ -1,6 +1,11 @@
-      Version: 1.2742.0
\ No newline at end of file
+      Version: 1.2742.0
+  OptionalDependencies:
```
JungleHelper is marked as an optional dependency here, but the code uses it as a hard dependency. Have you tested this with JungleHelper unloaded?

rhelmot commented on Code/SkinModHelperConfig.cs:
```diff
@@ -1,21 +1,48 @@
-        }
+    public class SkinModHelperConfig {
+        public string Options { get; set; }
+        public bool Player_List { get; set; }
```
It's generally bad form to use both CamelCase and underscores in the same identifier. Stick to CamelCase for C# public class variables.

rhelmot reviewed (commented):
context for bigkahuna: this review was solicited by aaa1459 over discord

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Update README.md

AAA1459 committed:
Delete mini_guide.txt

AAA1459 committed:
Update README.md

AAA1459 committed:
Add files via upload

AAA1459 committed:
Merge pull request #1 from AAA1459/backup

Trial new guide and some adjustment

AAA1459 committed:
Update README.md

AAA1459 committed:
Fixed a bug about colorGrade skin hair and feather

AAA1459 committed:
Compatible with UI-Mod "Save File Portraits"

Made the config skin's custom Portraits.xml compatible with "Save File Portraits"

Entering the map and return overworld After, skin's portraits.xml no longer affect save-file menu

AAA1459 committed:
Optimize register logic about skin's Portrait IDs

When registering skin's Portrait IDs: 
make "Path of IDs in Portraits.xmls" as encryption method instead of "Path of xmls own"

AAA1459 committed:
Fixed an accident error

AAA1459 committed:
Added features: any Skin of Free collocations

Options List"Extra Settings" rename to "General Skin"

About any Skin of Free collocations (Can On or Off): 
# it will list-out the reskin-able contect of general skin, then make you can to choose what kind will get what skin
# even is, let some kind only will get vanilla, or let some kind get general skin that low-priority
## (and changes in the mechanism for make them work, possible lack of feedback)

AAA1459 committed:
Make all hasHair-ID can get own bangs or hair

[SpecificPlayerSprite_Path]'s default value changed to “[SourcesPath value of Character_ID in sprites.xml]"

now, all hasHair-ID in Sprites.xml can get own bangs or hair, by use these paths:
# "../Gameplay/[IDself's SourcesPath]/bangs[number].png"
# "../Gameplay/[IDself's SourcesPath]/hair00.png"

AAA1459 committed:
Delete function_added.txt

AAA1459 committed:
Delete SkinSwapTrigger.cs

AAA1459 committed:
Fix some textbox bugs and adjust SkinSwapTrigger

Fix crash problem of the ask textboxes in intro cutscene of reflections
Fix the portraits skin cannot work for mini dialog

and adjust some other small issues?

AAA1459 committed:
fix some skin error work after changed setting

AAA1459 committed:
At bottom of guide, add about .mate.yaml flie

AAA1459 committed:
Update README.md

AAA1459 committed:
Fixed mirrormasks lost when using colorGrade skin

Fixed colorGrade skin error show in mirror
Fixed other Entity not show in mirror when using colorGrade skin

Debug: when player skin run missed animation, temporarily make player jump to madeline
